src/reducers/AddSiteReducer.js

```diff
@@ -74,14 +95,32 @@ export default (state = INITIAL_STATE, action) => {
             return {...state, facilitiesOption};
 
         case ADD_SITE_FIELDS_RESET:
-            return INITIAL_STATE;
+            return {...INITIAL_STATE, addSiteModalVisible: true};
 
         case CURRENT_LOCATION_UPDATED:
             const {currentLocation: {longitude, latitude}} = payload;
-            const stringLong =  _.toString(longitude);
+            const stringLong = _.toString(longitude);
             const stringLat = _.toString(latitude);
 
-            return {...state, longitudeText: stringLong, latitudeText: stringLat};
+            return {
+                ...state,
+                longitudeText: stringLong,
+                latitudeText: stringLat,
+                readyLatitude: latitude,
+                readyLongitude: longitude
+            };
+
+        case CHECK_IF_SITE_IS_READY:
+            const siteReadyForUpload = siteIsReadyForUpload(state);
+
+            return {...state, siteReadyForUpload};
+
+        case ADD_SITE_SUCCESS:
+            return INITIAL_STATE;
+
+        case ADD_SITE_FAILURE:
+            return {...state, addSiteModalVisible: false};
```

Comment:
When uploading a site works well, clear all `addSite` state (including the `addSiteModalVisible` which is initially set to `false`). If the site upload fails, KEEP all the state (that would REALLY suck if we threw everything away), but close the modal. @amgoodman5 again, we need to discuss what happens in both of these cases from a UX standpoint.